src/PaymentGateways/PayPalCommerce/Repositories/PayPalOrder.php

```diff
@@ -280,27 +281,59 @@ public function updateOrderAmount($orderId, array $array)
     }
 
     /**
+     * Update order amount using the Donation model.
+     *
+     * @unreleased updated to support donation category
      * @since 4.1.0 Add PayPal-Partner-Attribution-Id header
      * @since 4.0.0
      *
      * @throws Exception|HttpException|IOException
-     * @see        https://developer.paypal.com/docs/api/orders/v2/#orders_patch
+     * @see https://developer.paypal.com/docs/api/orders/v2/#orders_patch
      *
      */
-    public function updateApprovedOrder(string $orderId, Money $amount)
+    public function updateOrderFromDonation(string $orderId, Donation $donation)
     {
         $patchRequest = new OrdersPatchRequest($orderId);
 
         $patchRequest->headers["PayPal-Partner-Attribution-Id"] = give('PAYPAL_COMMERCE_ATTRIBUTION_ID');
 
+        $value = $donation->amount->formatToDecimal();
+        $currency = $donation->amount->getCurrency()->getCode();
+        $name = $donation->formTitle;
+
+        $amountParameters = [
+            'amount' => [
+                'value' => $value,
+                'currency_code' => $currency
+            ],
+        ];
+
+        if ($this->settings->isTransactionTypeDonation()) {
+            $amountParameters['items'] = [
+                [
+                    'name' => $name,
+                    'unit_amount' => [
+                        'value' => $value,
+                        'currency_code' => $currency,
+                    ],
+                    'quantity' => 1,
+                    'category' => 'DONATION',
+                ],
+            ];
+
+            $amountParameters['amount']['breakdown'] = [
+                'item_total' => [
+                    'currency_code' => $currency,
+                    'value' => $value,
+                ],
+            ];
+        }
+
```

Comment:
We have a similar logic inside the getAmountParameters() method, so I'm wondering if we can extract it to a separate method to be reused.